WMF Framework/Wikipedia.xcdatamodeld/Wikipedia 6.xcdatamodel/contents

```diff
@@ -73,6 +73,7 @@
     <entity name="TalkPage" representedClassName="TalkPage" syncable="YES" codeGenerationType="class">
         <attribute name="dateAccessed" attributeType="Date" defaultDateTimeInterval="582233760" usesScalarValueType="NO"/>
         <attribute name="displayTitle" attributeType="String"/>
+        <attribute name="forceRefresh" attributeType="Boolean" defaultValueString="NO" usesScalarValueType="YES"/>
```

Comment:
It seems a shame to have to add a permanent property to the model to handle a one-time migration.

Do you foresee other uses for this property?

If not I was wondering if it might make sense to do something a little sneaky like set the minutes and seconds of the dateAccessed to "33" minutes and "33" seconds (or '42' in honor of Douglas Adams), and set and check for those date components to refresh instead of the boolean.

It looks like you use the dateAccessed for cleanup purposes, but that shouldn't significantly affect sorting and maybe occasionally the rare talk page gets re-fetched unnecessarily.

Or I might be overthinking just to save a Bool.

Alternately, could there be a way to refetch as part of the migration?

Or - a user defaults key that just holds the appropriate record IDs, and they get checked and refreshed instead of the permanent bool on every record? (I am liking this second idea better.)

Reply:
(Or a list of revision IDs)

Reply:
@dempseyatgithub good ideas! Totally agreed on it feeling unnecessary - I don't foresee needing to keep track of a separate bool per talk page record in the future. I like the user defaults, I'll try that (though I must admit `dateAccessed` tinkering has the allure of not having an additional user defaults flag in the codebase). I'd rather not fetch on migration just because of the small chance that they don't have a connection in that case and/or the unnecessary traffic with fetching pages they may not intend to visit again.

Reply:
@dempseyatgithub I have posted a commit for using the user defaults method. I looked into `dateAccessed` first but found that it was being reset before the `forceRefresh` check ( within `userDidAccess`) which undid the seconds/nanoseconds specification I had made, and I didn't feel comfortable changing around the area of that reset at this point.